test/fake/masteruserrecord_client.go

```diff
@@ -0,0 +1,111 @@
+package fake
+
+import (
+	"encoding/json"
+	"os"
+
+	crtapi "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/testing"
+)
+
+type FakeMasterUserRecordClient struct {
+	Tracker   testing.ObjectTracker
+	Scheme    *runtime.Scheme
+	namespace string
+}
+
+func NewFakeMasterUserRecordClient(namespace string, initObjs ...runtime.Object) *FakeMasterUserRecordClient {
+	clientScheme := runtime.NewScheme()
+	err := crtapi.SchemeBuilder.AddToScheme(clientScheme)
+	if err != nil {
+		log.Error(err, "Error adding to scheme")
+		os.Exit(1)
+	}
+	crtapi.SchemeBuilder.Register(&crtapi.MasterUserRecord{}, &crtapi.MasterUserRecordList{})
+
+	tracker := testing.NewObjectTracker(clientScheme, scheme.Codecs.UniversalDecoder())
+	for _, obj := range initObjs {
+		err := tracker.Add(obj)
+		if err != nil {
+			log.Error(err, "failed to add object to fake user signup client", "object", obj)
+			panic("could not add object to tracker")
```

Comment:
Let's maybe add the original error message too?
```suggestion
			panic("could not add object to tracker: " + err.Error())
```

Reply:
Done